code/game/machinery/computer/_computer.dm

```diff
@@ -139,3 +144,37 @@
 			C.forceMove(loc)
 
 	qdel(src)
+
+/obj/machinery/computer/AltClick(mob/user)
+	if(!user.canUseTopic(src, issilicon(user)))
+		return
+	eject_id(user)
+
+/obj/machinery/computer/proc/insert_id(mob/user)
+	if(scan)
+		to_chat(user, "<span class='warning'>There's already an ID card in the console!</span>")
+		return
+	if(!scan)
+		var/obj/item/I = user.is_holding_item_of_type(/obj/item/card/id)
+		if(I)
+			if(!user.transferItemToLoc(I, src))
```

Comment:
just pass the attacking item as a param on insert_id

Reply:
I'm not sure what this means.